src/Container/Container.php

```diff
@@ -61,63 +60,13 @@ public function get($id): Extension
             ));
         }
 
-        $definition = $this->definitions[$id];
-
-        $service = $this->getService($id, $definition);
-
-        if (!$service instanceof Extension) {
-            throw new \RuntimeException(sprintf(
-                'Service resolved for identifier "%s" does not implement the %s" interface.',
-                $id,
-                Extension::class,
-            ));
-        }
+        $service = $this->definitions[$id]->resolve($this);
```

Comment:
We should probably check here, or better, in the constructor, whether we really have a `Container\Definition` here.